### PR TITLE
Remove credential file from spec

### LINKED ISSUE #206: Define local credential experience for Registry Authentication
Issue to track registry authentication experience in the absence of docker compatible credential provider in #192 . The proposed storage of unencrypted registry credential on local file system allows an attacker to potentially get read/write access to the registry.

Other options to consider, which arguably do not provide better security but gives the responsibility of securing the credentials to the end user, rather than in built support of unencrypted credentials in `notation`.
1. In the absence of a configured credential provider, require the user to provide user name and password as CLI arguments on every command that interacts with the registry (e.g. `notation sign` , `notation verify`)
2. Notation provides a minimal credential provider that takes input from environment variables (one option) as a script, end user can modify the script to pick credentials from file, or any other location they prefer.

I prefer option 1 for RC1, with addition of option 2 post RC1.

@SteveLasker , @sajayantony , @michaelb990 would be great to get your feedback.

Original content from PR #192
---

### Credential File

The credential file is alternative credential store when credential helpers are not available. The default file path is

```
{CONFIG}/notation/auth.json
```

The credential file path can be altered by setting the `credsFile` field of the notation config.

```json
{
    "credsFile": "/absolute/path/to/auth.json"
}
```

Since credentials are stored in plaintext, the permission of the credential file MUST be kept minimum when storing credentials. On Unix / Linux, the permission MUST be either `0600` (default) or `0400` (read-only).

### PR DESCRIPTION
**Pull request by shizhMSFT**
Resolve #206

### CODE REVIEW

File: specs/registry-auth.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ Notation follows the [Docker Registry v2 authentication][oauth2] specification f
 
 ## Credential Store
 
-As local credentials may be required to access the remote registries, they need to be stored and accessed securely. To achieve maximum security, credential helpers are preferred so that credentials are stored in the system key chain with better protection. If credential helpers are not available, notation will fall back to credential files with proper access control.
+As local credentials may be required to access the remote registries, they need to be stored and accessed securely. To achieve maximum security, credential helpers are preferred so that credentials are stored in the system key chain with better protection. If credential helpers are not available, credentials SHOULD be provided to notation via command line parameters `--username` / `--password` or environment variables `NOTATION_USERNAME` / `NOTATION_PASSWORD`.
```
Comment (binbin-li) [inline specs/registry-auth.md:98]: username/password options are only supported in `notation loging` cmd now. Should we mention it here?

Comment (dtzar) [reply specs/registry-auth.md:98]: Yes, we need to clarify what is or is not supported.

Comment (shizhMSFT) [reply specs/registry-auth.md:98]: username/password options are also supported for `notation sign`, `notation verify`, and other commands which need registry accesses.

**Review by gokarnm — APPROVED** [review]

**Review by binbin-li — APPROVED** [review]
lgtm

**Comment by FeynmanZhou** [comment]
Do we need to create another issue to update the [CLI baseline](https://github.com/notaryproject/notation/blob/main/specs/notation-cli.md#login) accordingly?

**Comment by binbin-li** [comment]
> Do we need to create another issue to update the [CLI baseline](https://github.com/notaryproject/notation/blob/main/specs/notation-cli.md#login) accordingly?

Yeah, I can make the code change to resolve it. Do we also need to update the cli spec?

**Review by FeynmanZhou — APPROVED** [review]

**Pull request merged**